apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/DatasetsResourceTest.java

```diff
@@ -4481,6 +4487,245 @@ private DatasetItemPage assertDatasetExperimentPage(UUID datasetId, UUID experim
+
+            String projectName = RandomStringUtils.randomAlphanumeric(20);
+
+            List<Trace> traces = createTrace().stream()
+                    .map(trace -> trace.toBuilder()
+                            .projectName(projectName)
+                            .build())
+                    .toList();
+
+            traces.parallelStream().forEach(trace -> createAndAssert(trace, workspaceName, apiKey));
+
+            var dataset = factory.manufacturePojo(Dataset.class);
+            var datasetId = createAndAssert(dataset, apiKey, workspaceName);
+
+            var datasetItemBatch = factory.manufacturePojo(DatasetItemBatch.class).toBuilder()
+                    .datasetId(datasetId)
+                    .build();
+
+            putAndAssert(datasetItemBatch, workspaceName, apiKey);
+
+            // Creating 5 different experiment ids
+            var experimentIds = IntStream.range(0, 5).mapToObj(__ -> GENERATOR.generate()).toList();
+
+            List<ExperimentItem> experimentItems = IntStream.range(0, 5)
+                    .mapToObj(i -> factory.manufacturePojo(ExperimentItem.class).toBuilder()
+                            .experimentId(experimentIds.get(i))
+                            .traceId(traces.get(i).id())
+                            .datasetItemId(datasetItemBatch.items().get(i).id()).build())
+                    .toList();
+
+            var experimentItemsBatch = ExperimentItemsBatch.builder()
+                    .experimentItems(Set.copyOf(experimentItems)).build();
+
+            createAndAssert(experimentItemsBatch, apiKey, workspaceName);
+
+            Set<Column> expectedOutput = getOutputDynamicColumns(traces);
+
+            assertColumns(datasetId, apiKey, workspaceName, null, expectedOutput);
+
+        }
+
+        @Test
+        void getExperimentItemsOutputColumns__whenFetchingByDatasetIdAndExperimentIds__thenReturnColumns() {
+            var workspaceName = UUID.randomUUID().toString();
+            var apiKey = UUID.randomUUID().toString();
+            var workspaceId = UUID.randomUUID().toString();
+
+            mockTargetWorkspace(apiKey, workspaceName, workspaceId);
+
+            String projectName = RandomStringUtils.randomAlphanumeric(20);
+
+            List<Trace> traces = createTrace().stream()
+                    .map(trace -> trace.toBuilder()
+                            .projectName(projectName)
+                            .build())
+                    .toList();
+
+            traces.parallelStream().forEach(trace -> createAndAssert(trace, workspaceName, apiKey));
+
+            var dataset = factory.manufacturePojo(Dataset.class);
+            var datasetId = createAndAssert(dataset, apiKey, workspaceName);
+
+            var datasetItemBatch = factory.manufacturePojo(DatasetItemBatch.class).toBuilder()
+                    .datasetId(datasetId)
+                    .build();
+
+            putAndAssert(datasetItemBatch, workspaceName, apiKey);
```

Comment:
all this part looks duplicated in both tests, if possible re-use

test is valid

Reply:
The setup looks like both create experiments. However, the distribution of traces between them is different. I will check what can be reused, but I believe not much will be reused because of those slight changes. Anyway, let me have a look